client/src/pages/SearchResultsPage.jsx

```diff
@@ -1,14 +1,151 @@
+import FiltersBar from "../components/FiltersBar";
 
 /**
  * https://github.com/BobaTalks/internship-tracker/issues/28
  * This page will display all the search results. Part of this work we should include the
  * search and location inputs.
  */
+export const FilterContext = createContext([{}, () => {}]);
+
 const SearchResultsPage = () => {
-  return <BasePage></BasePage>;
+  const [filterData, setFilterData] = useState({
+    remote: {
+      filterName: "Remote",
+      data: {
+        onsite: {
+          label: "On-site",
+          count: 4,
+          checked: false,
+        },
+        remote: {
+          label: "Remote",
+          count: 5,
+          checked: false,
+        },
+        hybrid: {
+          label: "Hybrid",
+          count: 12,
+          checked: false,
+        },
+      },
+    },
+    company: {
+      filterName: "Company",
+      data: {
+        meta: {
+          label: "Meta",
+          count: 4,
+          checked: false,
+        },
+        amazon: {
+          label: "Amazon",
+          count: 5,
+          checked: false,
+        },
+        netflix: {
+          label: "Netflix",
+          count: 15,
+          checked: false,
+        },
+        google: {
+          label: "Google",
+          count: 12,
+          checked: false,
+        },
+        apple: {
+          label: "Apple",
+          count: 2,
+          checked: false,
+        },
+      },
+    },
+    education: {
+      filterName: "Education",
+      data: {
+        currentStudent: {
+          label: "Current student",
+          count: 4,
+          checked: false,
+        },
+        recentGraduate: {
+          label: "Recent graduate",
+          count: 5,
+          checked: false,
+        },
+      },
+    },
+    semester: {
+      filterName: "Semester",
+      data: {
+        spring2023: {
+          label: "Spring 2023",
+          count: 4,
+          checked: false,
+        },
+        summer2023: {
+          label: "Summer 2023",
+          count: 5,
+          checked: false,
+        },
+        fall2023: {
+          label: "Fall 2023",
+          count: 10,
+          checked: false,
+        },
+        spring2024: {
+          label: "Spring 2024",
+          count: 23,
+          checked: false,
+        },
+        summer2024: {
+          label: "Summer 2024",
+          count: 8,
+          checked: false,
+        },
+      },
+    },
+    datePosted: {
+      filterName: "Date posted",
+      data: {
+        anyTime: {
+          label: "Any time",
+          count: 3,
+          checked: false,
+        },
+        pastDay: {
+          label: "Past 24 hours",
+          count: 2,
+          checked: false,
+        },
+        pastWeek: {
+          label: "Past week",
+          count: 21,
+          checked: false,
+        },
+        pastMonth: {
+          label: "Past month",
+          count: 12,
+          checked: false,
+        },
+      },
+    },
```

Comment:
is this just mock data? if so, we should probably pull this out into a constant and create a method to mock getting the data and supplying the state here

Reply:
can you elaborate?

Reply:
Can do something like 

```js
const MOCK_FILTER_DATA = {...}

const [filterData, setFilterData] = useState(MOCK_FILTER_DATA)
```

however, there will be logic to actually GET the available filters from the backend first right? We will have to create an empty object according to the available options.

pseudo logic might be something like

// fetch internships
// reduce internships to get dates posted, semesters, education, and company
// take this data to create keys in this object
// take this data to create counts in this object

we can revisit this once we get the endpoint though

but we can talk to @a-camarillo and figure out how the endpoint response would look like or expect to look like so we can start working on that logic as well, so we can write out the fetch logic still and just return a mock list of internships so we can extract this infromation

Reply:
Okay yep that makes sense

Reply:
Still brainstorming the endpoints for filtering, at the moment I'm not entirely sure if it's better design to have the filters be passed in as parameters to the jobs endpoint i.e. `GET /jobs?company_name=Google` or to have specific endpoints for the companies i.e. `GET /jobs/google`. 

Additionally there's some information that is being expected to be presented that I'm not entirely sure we can supply. Specifically, I know information related to date posted and semester isn't available on all company job posts. I suppose as a work around we can supply an N/A value so we're not eliminating the option from companies that do share the information.

Reply:
@a-camarillo We can filter the information in the frontend so you don't have to worry about passing parameters to the endpoint.

For information that can't be shown, can you communicate that to the design team? They can maybe determine if it's nicer to just completely remove the field or putting in a placeholder like N/A like you mentioned